psi4/src/psi4/scfgrad/scf_grad.h

```diff
@@ -71,6 +73,64 @@ class RSCFDeriv : public SCFDeriv {
     virtual SharedMatrix hessian_response() override;
 };
 
+class USCFDeriv : public SCFDeriv {
+protected:
+    std::shared_ptr<scf::UHF> uhf_wfn_;
+    SharedMatrix hessian_response_spin(int spin);
+    void overlap_deriv(std::shared_ptr<Matrix> C, 
+                              std::shared_ptr<Matrix> Cocc,
+                              std::shared_ptr<Matrix> Cvir,
+                              int nso, int nocc, int nvir, bool alpha);
+
+    void kinetic_deriv(std::shared_ptr<Matrix> C, 
+                       std::shared_ptr<Matrix> Cocc,
+                       int nso, int nocc, int nvir, bool alpha);
+
+    void potential_deriv(std::shared_ptr<Matrix> C, 
+                         std::shared_ptr<Matrix> Cocc,
+                         int nso, int nocc, int nvir, bool alpha);
+
+    void JK_deriv1(std::shared_ptr<Matrix> D1,
+                   std::shared_ptr<Matrix> C1, 
+                   std::shared_ptr<Matrix> C1occ,
+                   std::shared_ptr<Matrix> D2, 
+                   int nso, int nocc, int nvir, bool alpha);
+
+    void JK_deriv2(std::shared_ptr<JK> jk, int mem, 
+                   std::shared_ptr<Matrix> C1, 
+                   std::shared_ptr<Matrix> C1occ,
+                   std::shared_ptr<Matrix> C2, 
+                   std::shared_ptr<Matrix> C2occ,
+                   int nso, int n1occ, int n2occ, int nvir, bool alpha);
+
+    void VXC_deriv(std::shared_ptr<Matrix> C, 
+                   std::shared_ptr<Matrix> Cocc,
+                   int nso, int nocc, int nvir, bool alpha);
+
+    void assemble_Fock(int nocc, int nvir, bool alpha);
+
+    void assemble_B(std::shared_ptr<Vector> eocc, int nocc, int nvir, bool alpha);
+    void assemble_U(int nocc, int nvir, bool alpha);
+    void assemble_Q(std::shared_ptr<JK> jk,
+                    std::shared_ptr<Matrix> C1, 
+                    std::shared_ptr<Matrix> C1occ,
+                    std::shared_ptr<Matrix> C2, 
+                    std::shared_ptr<Matrix> C2occ, 
+                    int nso, int n1occ, int n2occ, int nvir,bool alpha);
+
+    void dipole_derivatives(std::shared_ptr<Matrix> C1, 
+                            std::shared_ptr<Matrix> C1occ,
+                            std::shared_ptr<Matrix> C2, 
+                            std::shared_ptr<Matrix> C2occ,
+                            std::shared_ptr<Matrix> Dt,
+                            int nso, int n1occ, int n2occ, int n1vir);
+
```

Comment:
These helpers are a good idea.  After this is in, I'll rework the RHF code to use them.